Code/Framework/AzFramework/Platform/Linux/AzFramework/Process/ProcessCommunicator_Linux.cpp

```diff
@@ -38,12 +68,40 @@ namespace AzFramework
     void CommunicatorHandleImpl::Close()
     {
         close(m_handle);
+        if (m_epollHandle != -1)
+        {
+            close(m_epollHandle);
+            m_epollHandle = -1;
+        }
         m_handle = -1;
     }
 
-    void CommunicatorHandleImpl::SetHandle(int handle)
+    bool CommunicatorHandleImpl::SetHandle(int handle, bool createEPollHandle)
     {
         m_handle = handle;
+        // also create an epoll handle
+        if ( (handle != -1) && (createEPollHandle) )
+        {
+            int m_epollHandle = epoll_create1(0);
+
+            AZ_Assert(m_epollHandle -1, "Failed to create epoll handle");
```

Comment:
may be worth it to add `errno` to the assert message here for troubleshooting
```suggestion
            AZ_Assert(m_epollHandle -1, "Failed to create epoll handle: error %d", errno);
```

Reply:
I'll add that, actually.